src/Mvc/Mvc.ViewFeatures/src/ViewComponentResultExecutor.cs

```diff
@@ -127,9 +138,26 @@ public virtual async Task ExecuteAsync(ActionContext context, ViewComponentResul
                 // IViewComponentHelper is stateful, we want to make sure to retrieve it every time we need it.
                 var viewComponentHelper = context.HttpContext.RequestServices.GetRequiredService<IViewComponentHelper>();
                 (viewComponentHelper as IViewContextAware)?.Contextualize(viewContext);
-
                 var viewComponentResult = await GetViewComponentResult(viewComponentHelper, _logger, result);
-                viewComponentResult.WriteTo(writer, _htmlEncoder);
+
+                if (viewComponentResult is ViewBuffer viewBuffer)
```

Comment:
https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.ViewFeatures/src/ViewComponents/DefaultViewComponentHelper.cs#L176 returns `ViewBuffer`. It's a bit unfortunate we have to downcast this since a user could have customized the implementation.

Reply:
Yeah I dunno why we made this stuff public. Sadbeans